cli/cli-client/src/main/java/org/infinispan/cli/commands/Migrate.java

```diff
@@ -0,0 +1,148 @@
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) {
+         // This command serves only to wrap the sub-commands
+         invocation.println(invocation.getHelpInfo());
+         return CommandResult.FAILURE;
+      }
+   }
+
+   @CommandDefinition(name = Migrate.ClusterConnect.CMD, description = "Connects to a source cluster", activator = DisabledActivator.class)
+   public static class ClusterConnect extends CliCommand {
+
+      public static final String CMD = "connect";
+
+      @Option(completer = CacheCompleter.class, shortName = 'c')
+      String cache;
+
+      @Option(shortName = 'u', description = "The Hot Rod URL for the source cluster", required = true)
+      String url;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      protected CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         throw new UnsupportedOperationException();
+      }
+   }
+
+   @CommandDefinition(name = Migrate.ClusterDisconnect.CMD, description = "Disconnects from a source cluster")
+   public static class ClusterDisconnect extends CliCommand {
+
+      public static final String CMD = "disconnect";
+
+      @Option(completer = CacheCompleter.class, shortName = 'c')
+      String cache;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      protected CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         CommandInputLine cmd = new CommandInputLine(Migrate.CMD)
+               .arg(Migrate.TYPE, Migrate.Cluster.CMD)
+               .arg(Migrate.SUBTYPE, ClusterDisconnect.CMD)
+               .option(CACHE, cache);
+         return invocation.execute(cmd);
+      }
+   }
+
+   @CommandDefinition(name = Migrate.ClusterSynchronize.CMD, description = "Synchronizes data from a source to a target cluster")
+   public static class ClusterSynchronize extends CliCommand {
+      public static final String CMD = "synchronize";
+      public static final String READ_BATCH = "read-batch";
+      public static final String THREADS = "threads";
+
+      @Option(completer = CacheCompleter.class, shortName = 'c')
+      String cache;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Option(shortName = 'b', description = "The amount of entries to process in a batch")
+      Integer readBatch;
+
+      @Option(shortName = 't', description = "The number of threads to use. Defaults to the number of cores on the server")
+      Integer threads;
```

Comment:
is there a way to disable the auto-completion for `threads` and `readBatch`?

If I do double-tab, I got the following exception:
```
[pedro-laptop-3-56000@cluster//containers/default]> migrate cluster synchronize --threads=May 20, 2020 12:21:36 PM org.aesh.command.impl.AeshCommandRuntime doSimpleComplete
SEVERE: Runtime exception when completing: Buffer: migrate cluster synchronize --threads=, Cursor:38, Offset:0, IgnoreOffset:false, Append separator: true, Candidates:[]
java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at org.aesh.command.impl.converter.IntegerConverter.convert(IntegerConverter.java:31)
	at org.aesh.command.impl.converter.IntegerConverter.convert(IntegerConverter.java:28)
	at org.aesh.command.impl.internal.ProcessedOption.doConvert(ProcessedOption.java:362)
	at org.aesh.command.impl.internal.ProcessedOption.injectValueIntoField(ProcessedOption.java:393)
	at org.aesh.command.impl.populator.AeshCommandPopulator.populateObject(AeshCommandPopulator.java:64)
	at org.aesh.command.impl.parser.AeshCommandLineCompletionParser.doInjectValues(AeshCommandLineCompletionParser.java:356)
	at org.aesh.command.impl.parser.AeshCommandLineCompletionParser.injectValuesAndComplete(AeshCommandLineCompletionParser.java:58)
	at org.aesh.command.impl.parser.AeshCommandLineParser.complete(AeshCommandLineParser.java:126)
	at org.aesh.command.impl.AeshCommandRuntime.doSimpleComplete(AeshCommandRuntime.java:332)
	at org.aesh.command.impl.AeshCommandRuntime.completeWithOperators(AeshCommandRuntime.java:278)
	at org.aesh.command.impl.AeshCommandRuntime.complete(AeshCommandRuntime.java:262)
	at org.aesh.readline.ReadlineConsole$AeshCompletion.complete(ReadlineConsole.java:409)
	at org.aesh.readline.ReadlineConsole$AeshCompletion.complete(ReadlineConsole.java:406)
	at org.aesh.readline.completion.CompletionHandler.createCompletionList(CompletionHandler.java:134)
	at org.aesh.readline.completion.CompletionHandler.complete(CompletionHandler.java:102)
	at org.aesh.readline.action.mappings.Complete.accept(Complete.java:62)
	at org.aesh.readline.Readline$AeshInputProcessor.parse(Readline.java:235)
	at org.aesh.readline.Readline$AeshInputProcessor.access$100(Readline.java:174)
	at org.aesh.readline.Readline.readInput(Readline.java:95)
	at org.aesh.readline.Readline.access$1000(Readline.java:57)
	at org.aesh.readline.Readline$AeshInputProcessor.lambda$start$1(Readline.java:324)
	at org.aesh.terminal.EventDecoder.accept(EventDecoder.java:118)
	at org.aesh.terminal.EventDecoder.accept(EventDecoder.java:31)
	at org.aesh.terminal.io.Decoder.write(Decoder.java:133)
	at org.aesh.readline.tty.terminal.TerminalConnection.openBlocking(TerminalConnection.java:216)
	at org.aesh.readline.tty.terminal.TerminalConnection.openBlocking(TerminalConnection.java:203)
	at org.aesh.readline.ReadlineConsole.accept(ReadlineConsole.java:243)
	at org.aesh.readline.ReadlineConsole.accept(ReadlineConsole.java:87)
	at org.aesh.readline.tty.terminal.TerminalConnection.init(TerminalConnection.java:153)
	at org.aesh.readline.tty.terminal.TerminalConnection.<init>(TerminalConnection.java:86)
	at org.aesh.readline.tty.terminal.TerminalConnection.<init>(TerminalConnection.java:96)
	at org.aesh.readline.ReadlineConsole.start(ReadlineConsole.java:183)
	at org.infinispan.cli.commands.CLI.interactive(CLI.java:203)
	at org.infinispan.cli.commands.CLI.exec(CLI.java:152)
	at org.infinispan.cli.commands.CliCommand.execute(CliCommand.java:32)
	at org.infinispan.cli.commands.CliCommand.execute(CliCommand.java:14)
	at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:213)
	at org.aesh.command.impl.AeshCommandRuntime.executeCommand(AeshCommandRuntime.java:161)
	at org.aesh.AeshRuntimeRunner.execute(AeshRuntimeRunner.java:118)
	at org.infinispan.cli.commands.CLI.main(CLI.java:252)
```

The only way I found to "fix" it was to create a converter:
```
   public Integer convert(ConverterInvocation converterInvocation) {
      String value = converterInvocation.getInput();
      if (value == null || value.isEmpty()) {
         return null;
      }
      return Integer.parseInt(value);
   }

```

Reply:
Added a NullableIntegerConverter